subscription.go

```diff
@@ -316,7 +316,11 @@ func (s *Subscription) Subscribe() error {
 	if !s.centrifuge.connected() {
 		return nil
```

Comment:
Here we return from `Subscribe` call, but since `needResubscribe` flag above set to true then at the moment Subscription will be subscribed as soon as connection restored. Now it won't work since we are adding it back to map later.

Reply:
Hm... Moving `storeSubscription` before `if !s.centrifuge.connected() {` would help?

Reply:
I think yes, could you please update this? If you can also add some tests – will be great. Hopefully, there are no other concerns - I'll try to look more closely during next week.

Reply:
Yes, I'll update today.